zerver/views/video_calls.py

```diff
@@ -151,3 +153,35 @@ def deauthorize_zoom_user(request: HttpRequest) -> HttpResponse:
             auth=(settings.VIDEO_ZOOM_CLIENT_ID, settings.VIDEO_ZOOM_CLIENT_SECRET),
         ).raise_for_status()
     return json_success()
+
+
+@zulip_login_required
+@never_cache
+def get_bbb_url(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+    id = "zulip-" + str(random.randint(100000000000, 999999999999))
+    password = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(10))
+    checksum = hashlib.sha1(("create" + "meetingID=" + id + "&moderatorPW=" + password + settings.BIG_BLUE_BUTTON_SECRET).encode()).hexdigest()
+    return json_success({"url": "/calls/bbb/join?meeting_id=%22" + id + "%22&password=%22" + password + "%22&checksum=%22" + checksum + "%22"})
```

Comment:
Same comment on URL-encoding as below.

Does this function need to compute the checksum?  It seems like it'd be better to do it in the `join` code method, if nothing else to simplify the URLs here.

Reply:
I generate the checksum here so that it is not possible to use zulip to get around the checksum security as otherwise it would be possible to join every meeting on the Big Blue Button Server by id and password

Reply:
Ahh, I see, I was skimming in this phase and hadn't noticed the checksum required a secret to generate.  OK.